src/main/java/uk/gov/hmcts/reform/slc/health/FtpHealthIndicator.java

```diff
@@ -1,31 +1,38 @@
 package uk.gov.hmcts.reform.slc.health;
 
-import org.springframework.boot.actuate.health.AbstractHealthIndicator;
 import org.springframework.boot.actuate.health.Health;
+import org.springframework.boot.actuate.health.HealthIndicator;
 import org.springframework.stereotype.Component;
 import uk.gov.hmcts.reform.slc.services.FtpAvailabilityChecker;
 import uk.gov.hmcts.reform.slc.services.steps.sftpupload.FtpClient;
 
 import java.time.LocalTime;
 
 @Component
-public class FtpHealthIndicator extends AbstractHealthIndicator {
+public class FtpHealthIndicator implements HealthIndicator {
 
     private final FtpAvailabilityChecker availabilityChecker;
     private final FtpClient client;
 
-    public FtpHealthIndicator(FtpAvailabilityChecker availabilityChecker,
-                              FtpClient client) {
+    public FtpHealthIndicator(
+        FtpAvailabilityChecker availabilityChecker,
+        FtpClient client
+    ) {
         this.availabilityChecker = availabilityChecker;
         this.client = client;
     }
 
     @Override
-    protected void doHealthCheck(Health.Builder builder) throws Exception {
-        if (availabilityChecker.isFtpAvailable(LocalTime.now()) && !client.isHealthy()) {
-            builder.down();
+    public Health health() {
+        if (!availabilityChecker.isFtpAvailable(LocalTime.now())) {
+            return Health.up().build();
         } else {
-            builder.up();
+            try {
+                client.testConnection();
+                return Health.up().build();
+            } catch (Exception exc) {
+                return Health.down().withException(exc).build();
+            }
```

Comment:
that is all dealt in abstract class hence the benefit of using it. `doHealthCheck` becomes a lot simpler:
```java
if (isFtpAvailable) {
    client.testConnection();
}

builder.up();
```

very clear, outcome is the same